src/kOS/AddOns/InfernalRobotics/IRControlGroup.cs

```diff
@@ -0,0 +1,87 @@
+﻿using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe.Utilities;
+using kOS.Suffixed;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using kOS.AddOns.InfernalRobotics;
+
+namespace kOS.AddOns.InfernalRobotics
+{
+    public class IRControlGroupWrapper : Structure
+    {
+        private readonly IRWrapper.IControlGroup cg;
+        private readonly SharedObjects shared;
+
+        public IRControlGroupWrapper(IRWrapper.IControlGroup init, SharedObjects shared)
+        {
+            cg = init;
+            this.shared = shared;
+            InitializeSuffixes();
+        }
+
+        private void InitializeSuffixes()
+        {
+            AddSuffix("NAME", new SetSuffix<string>(() => cg.Name, value => cg.Name = value));
+            AddSuffix("SPEED", new SetSuffix<float>(() => cg.Speed, value => cg.Speed = value));
+            AddSuffix("EXPANDED", new SetSuffix<bool>(() => cg.Expanded, value => cg.Expanded = value));
+            AddSuffix("FORWARDKEY", new SetSuffix<string>(() => cg.ForwardKey, value => cg.ForwardKey = value));
```

Comment:
oh man, this is pretty exciting. I can just imagine users being able to swap in/out different servos/groups live for some of advanced fly by wire. cool!